core/src/main/java/org/apache/iceberg/GenericManifestFile.java

```diff
@@ -481,14 +503,15 @@ private CopyBuilder(ManifestFile toCopy) {
                 toCopy.sequenceNumber(),
                 toCopy.minSequenceNumber(),
                 toCopy.snapshotId(),
+                copyList(toCopy.partitions(), PartitionFieldSummary::copy),
+                toCopy.keyMetadata(),
                 toCopy.addedFilesCount(),
                 toCopy.addedRowsCount(),
                 toCopy.existingFilesCount(),
                 toCopy.existingRowsCount(),
                 toCopy.deletedFilesCount(),
                 toCopy.deletedRowsCount(),
-                copyList(toCopy.partitions(), PartitionFieldSummary::copy),
```

Comment:
Is there a reason the reordering of parameters here? It's not immediately apparent to me.

Reply:
Yes. An earlier commit added a new non-public constructor with reordered arguments. I'm modifying that one and deprecating the public constructor.